src/gallery.js

```diff
@@ -0,0 +1,122 @@
+			previousButton = galleryContainer.querySelectorAll( '.wp-gallery-popup-button.previous' )[ 0 ],
+			next = current + 1
+
+		if ( gallery[ next ] ) {
+			image.src = gallery[ next ].src
+			caption.innerHTML = gallery[ next ].caption ? gallery[ next ].caption : ''
+			nextButton.style.opacity = gallery[ next + 1 ] ? '1' : '0.5'
+			previousButton.style.opacity = current === 0 ? '1' : null
+			current += 1
+		}
+	},
+
+	renderPrevious = ( galleryContainer ) => {
+		const image = galleryContainer.querySelector( '.wp-gallery-popup-image' ).firstChild,
+			caption = galleryContainer.querySelector( '.wp-gallery-popup-caption' ),
+			previousButton = galleryContainer.querySelectorAll( '.wp-gallery-popup-button.previous' )[ 0 ],
+			nextButton = galleryContainer.querySelectorAll( '.wp-gallery-popup-button.next' )[ 0 ],
+			previous = current - 1
+
+		if ( gallery[ previous ] ) {
+			image.src = gallery[ previous ].src
+			caption.innerHTML = gallery[ previous ].caption ? gallery[ previous ].caption : ''
+			previousButton.style.opacity = gallery[ previous - 1 ] ? '1' : '0.5'
+			nextButton.style.opacity = current === gallery.leght - 1 ? '1' : null
+			current -= 1
+		}
+	},
+
+	showFullscreenGallery = ( event, mediaItems, popup, doc = document ) => {
+		const fullscreenGallery = doc.createElement( 'div' ),
+			closeButton = doc.createElement( 'div' ),
+			nextButton = doc.createElement( 'div' ),
+			previousButton = doc.createElement( 'div' ),
+			imageContainer = doc.createElement( 'div' ),
+			image = doc.createElement( 'img' ),
+			caption = doc.createElement( 'p' ),
+			src = event.target.style.backgroundImage.slice( 4, -1 ).replace( /"/g, '' ),
+			galleryContainer = popup.element.querySelector( '.wikipediapreview-gallery' )
+
+		gallery = mediaItems
+		gallery.forEach( ( image, index ) => {
+			if ( image.src === src ) {
+				current = index
+			}
+		} )
+
+		fullscreenGallery.classList.add( 'wp-gallery-popup' )
+
+		image.src = src
+		imageContainer.appendChild( image )
+		imageContainer.classList.add( 'wp-gallery-popup-image' )
+
+		caption.innerHTML = gallery[ current ].caption ? gallery[ current ].caption : ''
+		caption.classList.add( 'wp-gallery-popup-caption' )
+
+		closeButton.classList.add( 'wp-gallery-popup-button', 'close' )
+		closeButton.addEventListener( 'click', () => {
+			hideFullscreenGallery( galleryContainer )
+		} )
+		nextButton.classList.add( 'wp-gallery-popup-button', 'next' )
+		nextButton.style.opacity = current === gallery.length - 1 ? '0.5' : '1'
+		nextButton.addEventListener( 'click', () => {
+			renderNext( galleryContainer )
+		} )
+		previousButton.classList.add( 'wp-gallery-popup-button', 'previous' )
+		previousButton.style.opacity = current === 0 ? '0.5' : '1'
+		previousButton.addEventListener( 'click', () => {
+			renderPrevious( galleryContainer )
+		} )
+
+		if ( gallery.length > 1 ) {
+			Array.prototype.forEach.call(
```

Comment:
Why do you use this syntax instead of `[...].forEach(...)`?

Reply:
Hmm I thought there was a browser compatibility issue but apparently not, `[...].forEach(...)` seems to be working fine and it's more readable. Updating...